cpp/src/arrow/adapters/orc/util.h

```diff
@@ -23,6 +23,7 @@
 #include "arrow/array/builder_base.h"
 #include "arrow/result.h"
 #include "arrow/status.h"
+#include "arrow/util/key_value_metadata.h"
```

Comment:
Only the forward declarations should be necessary here.
```suggestion
#include "arrow/type_fwd.h"
```